src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs

```diff
@@ -420,24 +465,32 @@ private async Task<OpenApiOperation> GenerateOpenApiOperationFromMetadataAsync(A
         }
 
         // Schemas will be generated via Swashbuckle by default.
-        foreach (var parameter in operation.Parameters)
+        if (operation.Parameters is { Count: > 0 } parameters)
         {
-            var apiParameter = apiDescription.ParameterDescriptions.SingleOrDefault(desc => desc.Name == parameter.Name && !desc.IsFromBody() && !desc.IsFromForm() && !desc.IsIllegalHeaderParameter());
-            if (apiParameter is not null)
+            foreach (var parameter in parameters)
```

Comment:
I haven't tried, but I _think_ this could be written as: `foreach (var parameter in parameters ?? [])`, to reduce the nesting.

Reply:
I haven't tried using C# 14-isms yet (and I think I'd need to opt in to preview language version).

Reply:
I don't understand, the `??` operator is very old and used in lots of other places already.

Reply:
`?? []` is the part I'm referring to when used within a foreach. It typically doesn't work as it doesn't know the target type to use.

Reply:
Actually, I think I'd confused it with https://github.com/dotnet/csharplang/issues/7913